SwiftUIDZ/ModelView/AuthenticationViewModel.swift

```diff
@@ -44,14 +57,22 @@ final class AuthenticationViewModel: ObservableObject {
         }
     }
 
-    func getResultUserCheck() -> String {
-        if
-            Constants.minPasswordNumber ... Constants.maxPasswordNumber ~= lastPasswordText.count,
-            lastNumberText.count == Constants.phoneNumber
-        {
-            return Constants.detailViewTagText
-        } else {
-            return Constants.emptyText
-        }
+    func progressViewActivate(completion: @escaping (String) -> Void) {
+        Timer.scheduledTimer(withTimeInterval: Constants.timeInterval, repeats: true, block: { timer in
+            if self.progressViewCount == Constants.progressMaxCount {
```

Comment:
```suggestion
            if self.progressViewCount == Constants.progressMaxCount {
```
замени большинство if на guard